src/analysis/retail/mage/fire/core/HotStreak.tsx

```diff
@@ -82,47 +88,81 @@ class HotStreak extends Analyzer {
     this.wasted.push({ event: event, timestamp: event.timestamp });
   }
 
+  onFightEnd(event: FightEndEvent) {
+    this.analyzeHotStreaks();
+  }
+
   // prettier-ignore
-  missingHotStreakPreCast = () => {
+  analyzeHotStreaks = () => {
+
     //If there is a precast then filter it out. 
-    let missedCasts = this.hotStreaks.filter(hs => !hs.precast)
+    let procs = this.hotStreaks.filter(hs => {
+      if (hs.precast) {
+        hs.preCastMissing = { value: QualitativePerformance.Good, tooltip: `Precast ${hs.precast?.ability.name} Found` }  
+      }
+      return !hs.preCastMissing;
+    })
 
-    //If Combustion or Hyperthermia was active, filter it out
-    missedCasts = missedCasts.filter(hs => {
+    //If Ccombustion was active, filter it out
+    procs = procs.filter(hs => {
       const combustionActive = this.selectedCombatant.hasBuff(TALENTS.COMBUSTION_TALENT.id, hs.remove.timestamp);
-      const hyperthermiaActive = this.selectedCombatant.hasBuff(SPELLS.HYPERTHERMIA_BUFF.id, hs.remove.timestamp);
-      return (!this.hasHyperthermia || !hyperthermiaActive) && (!combustionActive)
+      if (combustionActive) {
+        hs.preCastMissing = { value: QualitativePerformance.Good, tooltip: `Combustion Active`}
```

Comment:
I'm assuming each assignment to `hs.preCastMissing` precludes the remaining checks from reassigning to `hs.preCastMissing`? If not, perhaps it's worth showing each item as it's own check. E.g. one line for each of the values.

![image](https://github.com/WoWAnalyzer/WoWAnalyzer/assets/6109344/bae58d78-6434-44cd-886c-00da3f5811d8)

I'd be interested in other opinions but I would have thought `QualitativePerformance.Perfect` would be preferred over `Good` for consistency with other checks.

Reply:
Yes each check filters out the ones it already assigned a performance thing to.

I tended towards using good for stuff they did properly and perfect for stuff they did great (generally whenever they are under the minor threshold). I might change it on my next pass but for now I'll leave it